### DISCUSSION TITLE
[Enhancement] [menu] el-menu没有只保持一个子菜单的展开的属性：unique-opened

### DISCUSSION THREAD

**Discussion by viastronger** [element-plus/element-plus#18815] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `menu`

## Description

老版的element，el-menu有个属性是unique-opened：是否只保持一个子菜单的展开，element-plus没有这个属性，有些时候还是挺有用的，建议加上

### Additional comments

_(empty)_

**Comment by warmthsea** [comment]
element-plus 同样有这个属性

https://staging.element-plus.org/zh-CN/component/menu.html#menu-attributes

[Playground Demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuIFxuICA8ZWwtbWVudVxuICAgIGRlZmF1bHQtYWN0aXZlPVwiMlwiXG4gICAgY2xhc3M9XCJlbC1tZW51LXZlcnRpY2FsLWRlbW9cIlxuICAgIHVuaXF1ZS1vcGVuZWRcbiAgICBAb3Blbj1cImhhbmRsZU9wZW5cIlxuICAgIEBjbG9zZT1cImhhbmRsZUNsb3NlXCJcbiAgPlxuICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjFcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+XG4gICAgICAgIDxlbC1pY29uPjxsb2NhdGlvbiAvPjwvZWwtaWNvbj5cbiAgICAgICAgPHNwYW4+TmF2aWdhdG9yIE9uZTwvc3Bhbj5cbiAgICAgIDwvdGVtcGxhdGU+XG4gICAgICA8ZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgICAgICA8dGVtcGxhdGUgI3RpdGxlPjxzcGFuPkdyb3VwIE9uZTwvc3Bhbj48L3RlbXBsYXRlPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMS0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8L2VsLW1lbnUtaXRlbS1ncm91cD5cbiAgICAgIDxlbC1tZW51LWl0ZW0tZ3JvdXAgdGl0bGU9XCJHcm91cCBUd29cIj5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjEtM1wiPml0ZW0gdGhyZWU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgICAgPGVsLXN1Yi1tZW51IGluZGV4PVwiMS00XCI+XG4gICAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+PHNwYW4+aXRlbSBmb3VyPC9zcGFuPjwvdGVtcGxhdGU+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIxLTQtMVwiPml0ZW0gb25lPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8L2VsLXN1Yi1tZW51PlxuICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjJcIj5cbiAgICAgIDxlbC1pY29uPjxpY29uLW1lbnUgLz48L2VsLWljb24+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPk5hdmlnYXRvciBUd288L3RlbXBsYXRlPlxuICAgIDwvZWwtbWVudS1pdGVtPlxuICAgICA8ZWwtc3ViLW1lbnUgaW5kZXg9XCI1XCI+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPlxuICAgICAgICA8ZWwtaWNvbj48bG9jYXRpb24gLz48L2VsLWljb24+XG4gICAgICAgIDxzcGFuPk5hdmlnYXRvciA1PC9zcGFuPlxuICAgICAgPC90ZW1wbGF0ZT5cbiAgICAgIDxlbC1tZW51LWl0ZW0tZ3JvdXA+XG4gICAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+PHNwYW4+R3JvdXAgT25lPC9zcGFuPjwvdGVtcGxhdGU+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI1LTFcIj5pdGVtIG9uZTwvZWwtbWVudS1pdGVtPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNS0yXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtbWVudS1pdGVtLWdyb3VwPlxuICAgIDwvZWwtc3ViLW1lbnU+XG4gIDwvZWwtbWVudT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQge1xuICBEb2N1bWVudCxcbiAgTWVudSBhcyBJY29uTWVudSxcbiAgTG9jYXRpb24sXG4gIFNldHRpbmcsXG59IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBpc0NvbGxhcHNlID0gcmVmKCF0cnVlKVxuY29uc3QgaGFuZGxlT3BlbiA9IChrZXk6IHN0cmluZywga2V5UGF0aDogc3RyaW5nW10pID0+IHtcbiAgY29uc29sZS5sb2coa2V5LCBrZXlQYXRoKVxufVxuY29uc3QgaGFuZGxlQ2xvc2UgPSAoa2V5OiBzdHJpbmcsIGtleVBhdGg6IHN0cmluZ1tdKSA9PiB7XG4gIGNvbnNvbGUubG9nKGtleSwga2V5UGF0aClcbn1cbjwvc2NyaXB0PlxuXG48c3R5bGU+XG4uZWwtbWVudS12ZXJ0aWNhbC1kZW1vOm5vdCguZWwtbWVudS0tY29sbGFwc2UpIHtcbiAgd2lkdGg6IDIwMHB4O1xuICBtaW4taGVpZ2h0OiA0MDBweDtcbn1cbjwvc3R5bGU+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L3RoZW1lLWNoYWxrL2RhcmsvY3NzLXZhcnMuY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn1cbiIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9cIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6e319)
